scripts/init.sh

```diff
@@ -77,24 +77,26 @@ if [ -r $DEFAULT ]; then
     source $DEFAULT
 fi
 
-function pidofproc() {
-    if [ $# -ne 3 ]; then
-        echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
-    fi
+if ! typeset -F pidofproc &>/dev/null; then
+  function pidofproc() {
+      if [ $# -ne 3 ]; then
+          echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
+      fi
 
-    PID=`pgrep -f $3`
-    local PIDFILE=`cat $2`
+      pid=`pgrep -f '\'$3'\>'`
```

Comment:
OK, thanks. I'm leaning towards the simpler, clearer option unless you can provide some references. This is an important script obviously, so we need to be careful with what we do.